handlers.go

```diff
@@ -366,13 +365,13 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 	defer func() { logger.FinishWithErr(ctx, _err) }()
 	logger.SetTag(ctx, "peer", p)
 
-	c, err := cid.Cast([]byte(pmes.GetKey()))
-	if err != nil {
-		return nil, err
+	key := pmes.GetKey()
+	if len(key) > 80 {
+		return nil, fmt.Errorf("handleAddProviders key size too large")
```

Comment:
why this check?

Reply:
80 bytes -> 512 bits + change. It's enough for the largest reasonable hash plus some metadata. We don't want arbitrarily large keys to be used as a DoS vector.

If your keys are larger, you should double hash them.